conans/client/generators/pkg_config.py

```diff
@@ -18,54 +20,69 @@
 """
 
 
-def concat_if_not_empty(groups):
-    return " ".join([param for group in groups for param in group if param and param.strip()])
-
-
-def single_pc_file_contents(name, cpp_info):
-    lines = ['prefix=%s' % cpp_info.rootpath.replace("\\", "/")]
-    libdir_vars = []
-    for i, libdir in enumerate(cpp_info.libdirs):
-        varname = "libdir" if i == 0 else "libdir%d" % (i + 2)
-        lines.append("%s=${prefix}/%s" % (varname, libdir))
-        libdir_vars.append(varname)
-    include_dir_vars = []
-    for i, includedir in enumerate(cpp_info.includedirs):
-        varname = "includedir" if i == 0 else "includedir%d" % (i + 2)
-        lines.append("%s=${prefix}/%s" % (varname, includedir))
-        include_dir_vars.append(varname)
-    lines.append("")
-    lines.append("Name: %s" % name)
-    description = cpp_info.description or "Conan package: %s" % name
-    lines.append("Description: %s" % description)
-    lines.append("Version: %s" % cpp_info.version)
-    libdirs_flags = ["-L${%s}" % name for name in libdir_vars]
-    libnames_flags = ["-l%s " % name for name in cpp_info.libs]
-    shared_flags = cpp_info.sharedlinkflags + cpp_info.exelinkflags
-    lines.append("Libs: %s" % concat_if_not_empty([libdirs_flags, libnames_flags, shared_flags]))
-    include_dirs_flags = ["-I${%s}" % name for name in include_dir_vars]
-
-    lines.append("Cflags: %s" % concat_if_not_empty(
-                                 [include_dirs_flags,
-                                  cpp_info.cppflags,
-                                  cpp_info.cflags,
-                                  ["-D%s" % d for d in cpp_info.defines]]))
-
-    if cpp_info.public_deps:
-        public_deps = " ".join(cpp_info.public_deps)
-        lines.append("Requires: %s" % public_deps)
-    return "\n".join(lines) + "\n"
-
-
 class PkgConfigGenerator(Generator):
 
     @property
     def filename(self):
         pass
 
+    @property
+    def compiler(self):
+        return self.conanfile.settings.get_safe("compiler")
+
     @property
     def content(self):
         ret = {}
         for depname, cpp_info in self.deps_build_info.dependencies:
-            ret["%s.pc" % depname] = single_pc_file_contents(depname, cpp_info)
+            ret["%s.pc" % depname] = self.single_pc_file_contents(depname, cpp_info)
         return ret
+
+    def single_pc_file_contents(self, name, cpp_info):
+        lines = ['prefix=%s' % cpp_info.rootpath.replace("\\", "/")]
+        libdir_vars = []
+        for i, libdir in enumerate(cpp_info.libdirs):
+            libdir = libdir.replace("\\", "/")
+            varname = "libdir" if i == 0 else "libdir%d" % (i + 2)
+            prefix = "${prefix}/" if self._with_prefix(libdir) else ""
```

Comment:
Having abs paths non-prefixes makes no sense here. The prefix is to be able to replace the base path and reuse the package on another computer. So, is "libdirs" finally giving abs paths? how could we get relative ones? If the user is appending an absolute one in the recipe, we need to do:

`libdir = os.path.relpath(cpp_info.rootpath, libdir)`

Reply:
Yes, you are right. However if you check the issue you will see that `boost_math` package has lib dirs pointing to a different path (actually pointing to the lib path of another package) and `cpp_info.rootpath` is not a prefix of libdir path.

And I think this also happens in Windows when `short_paths=True`.